spec/datagrid/filters_spec.rb

```diff
@@ -292,5 +292,33 @@ class TestGrid8728
       expect(non_admin_filters).to_not include(:id)
       expect(non_admin_filters).to include(:name)
     end
+
+    context 'with delegation to attribute' do
+      let(:role) { OpenStruct.new('admin?' => admin) }
+      let(:klass) do
+        test_report_class do
+          attr_accessor :role
+          delegate :admin?, to: :role
+
+          scope { Entry }
+
+          filter(:id, :integer, if: :admin?)
+        end
+      end
+
+      subject { klass.new(role: role).filters.map(&:name) }
```

Comment:
Are you sure your test fails when there is no patch? You've only fixed the default value assignment in constructor. I think you should test for that.